src/main/java/ua/realestate/management/security/service/SecurityUserService.java

```diff
@@ -0,0 +1,37 @@
+package ua.realestate.management.security.service;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+import ua.realestate.management.entity.UserEntity;
+import ua.realestate.management.repository.UserRepository;
+import ua.realestate.management.security.entity.SecurityUserEntity;
+
+import javax.transaction.Transactional;
+
+import static ua.realestate.management.exception.ErrorMessage.USER_NOT_FOUND_BY_ID;
+
+@Service
+@RequiredArgsConstructor
+public class SecurityUserService implements UserDetailsService {
+    private final UserRepository userRepository;
+
+
+    @Override
+    @Transactional
+    public UserDetails loadUserByUsername(String email) throws UsernameNotFoundException {
```

Comment:
If the email is really a username in your case (and the name is just a custom name), then it's okay. Still, if you would also need to load users by their names, it might be a problem, and I would suggest loading users by names and adding additional validation (when needed).

Reply:
The email should be unique.
Like in real life: it happens, that there are people with the same first name and last name. It's impossible, that two peoples have the same email address.

Reply:
Well, **username** is not the same as first name or last name. Usually, it's a combination of unique symbols that are using for login to the systems, it might be email address as well, but in this case, I'm not sure that you need both fields in the database unless you are going to let the users choose another email-box for delivering notifications.